<p>Public modules should include a <code>@moduledoc</code> attribute describing their purpose.</p>

<h2>Noncompliant Code Example</h2>
<pre>
defmodule MyApp.Accounts do
  def list_users do
    Repo.all(User)
  end
end
</pre>

<h2>Compliant Solutions</h2>
<pre>
defmodule MyApp.Accounts do
  @moduledoc """
  Context for managing user accounts.
  """

  def list_users do
    Repo.all(User)
  end
end
</pre>

<p>To intentionally suppress the warning for internal or private modules:</p>
<pre>
defmodule MyApp.Accounts.Helpers do
  @moduledoc false

  # ...
end
</pre>

<h2>See</h2>
<ul>
  <li><a href="https://hexdocs.pm/elixir/writing-documentation.html">Elixir - Writing Documentation</a></li>
  <li>Credo: <code>Credo.Check.Readability.ModuleDoc</code></li>
</ul>
