<p>Modules that grow too large become difficult to understand, test, and maintain. Consider splitting large modules into smaller, focused modules that each represent a single concept.</p>

<h2>Noncompliant Code Example</h2>
<pre>
defmodule MyApp.KitchenSink do
  # 300+ lines of mixed concerns
end
</pre>

<h2>Compliant Solution</h2>
<pre>
defmodule MyApp.Accounts do
  # User account logic
end

defmodule MyApp.Notifications do
  # Notification delivery logic
end
</pre>

<h2>Configuration</h2>
<p>The default threshold is 500 lines per module, matching the Erlang community convention (Elvis linter).</p>

<h2>See Also</h2>
<ul>
  <li>Elixir style guides recommend keeping modules focused on a single concept</li>
</ul>
